server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java

```diff
@@ -118,6 +125,206 @@ public void testCreateSearchableSnapshot() throws Exception {
+        assertDocCount(restoredIndexName, 100L);
+
+        logger.info("--> add 2 new search nodes");
+        internalCluster().ensureAtLeastNumSearchNodes(numReplicasIndex + 2);
+        ensureGreen(restoredIndexName);
+        assertDocCount(restoredIndexName, 100);
+
+        logger.info("--> stop a random search node");
+        internalCluster().stopRandomSearchNode();
+        ensureGreen(restoredIndexName);
+        assertDocCount(restoredIndexName, 100);
+
+    }
+
+    /**
+     * Tests the functionality of {@link org.opensearch.cluster.routing.allocation.allocator.RemoteShardsBalancer} to
+     * when no {@link org.opensearch.cluster.node.DiscoveryNodeRole#SEARCH_ROLE} capable nodes are present in the cluster.
+     */
+    public void testSearchableSnapshotAllocationWithNoSearchNodes() throws Exception {
+        final int numReplicasIndex = randomIntBetween(1, 4);
+        final String indexName = "test-idx";
+        final String restoredIndexName = indexName + "-copy";
+        internalCluster().ensureAtLeastNumDataNodes(numReplicasIndex + 1);
+        final Client client = client();
+        createRepository("test-repo", "fs");
+        createIndex(
+            indexName,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, Integer.toString(numReplicasIndex))
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, "1")
+                .build()
+        );
+        ensureGreen();
+
+        indexRandomDocs(indexName, 100);
+        ensureGreen();
+
+        logger.info("--> Take a snapshot");
+        final CreateSnapshotResponse createSnapshotResponse = client.admin()
+            .cluster()
+            .prepareCreateSnapshot("test-repo", "test-snap")
+            .setWaitForCompletion(true)
+            .setIndices(indexName)
+            .get();
+        MatcherAssert.assertThat(createSnapshotResponse.getSnapshotInfo().successfulShards(), greaterThan(0));
+        MatcherAssert.assertThat(
+            createSnapshotResponse.getSnapshotInfo().successfulShards(),
+            equalTo(createSnapshotResponse.getSnapshotInfo().totalShards())
+        );
+
+        logger.info("--> delete local index");
+        assertTrue(client.admin().indices().prepareDelete(indexName).get().isAcknowledged());
+        ensureGreen();
+
+        logger.info("--> restore index as 'remote_snapshot' with no search capable nodes waiting for completion");
+        // No search nodes within the cluster will lead to the request never completing
+        // which is consistent with file-system restore with wait for complete.
+        assertThrows(OpenSearchTimeoutException.class, () -> client.admin()
+            .cluster()
+            .prepareRestoreSnapshot("test-repo", "test-snap")
+            .setRenamePattern("(.+)")
+            .setRenameReplacement("$1-copy")
+            .setStorageType(RestoreSnapshotRequest.StorageType.REMOTE_SNAPSHOT)
+            .setWaitForCompletion(true)
```

Comment:
I'd be inclined to not set "wait for completion" because you assert the relevant state below and it is usually best to avoid making tests wait on timeouts like this if possible

Reply:
Got rid of the synchronous timeout based call.